From d0b45a488f40f082493a53fe49e3d85c6d6f0762 Mon Sep 17 00:00:00 2001
From: Andrew Purtell <apurtell@apache.org>
Date: Thu, 28 Aug 2014 15:53:19 -0700
Subject: [PATCH] HBASE-11846 HStore#assertBulkLoadHFileOk should log if a full
 HFile verification will be performed during a bulkload (Esteban Gutierrez)

Amending-Author: Andrew Purtell <apurtell@apache.org>
---
 .../src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java   | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java
index 7cb5dc2..f2c86e1 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java
@@ -642,6 +642,8 @@ public class HStore implements Store {
       }
 
       if (verifyBulkLoads) {
+        long verificationStartTime = EnvironmentEdgeManager.currentTimeMillis();
+        LOG.info("Full verification started for bulk load hfile: " + srcPath.toString());
         KeyValue prevKV = null;
         HFileScanner scanner = reader.getScanner(false, false, false);
         scanner.seekTo();
@@ -667,6 +669,9 @@ public class HStore implements Store {
           }
           prevKV = kv;
         } while (scanner.next());
+      LOG.info("Full verification complete for bulk load hfile: " + srcPath.toString()
+         + " took " + (EnvironmentEdgeManager.currentTimeMillis() - verificationStartTime)
+         + " ms");
       }
     } finally {
       if (reader != null) reader.close();
-- 
1.8.5.2 (Apple Git-48)

